test/PowerShellEditorServices.Test.E2E/Processes/StdioServerProcess.cs

```diff
@@ -0,0 +1,137 @@
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace PowerShellEditorServices.Test.E2E
+{
+    /// <summary>
+    ///     A <see cref="StdioServerProcess"/> is a <see cref="ServerProcess"/> that launches its server as an external process and communicates with it over STDIN / STDOUT.
+    /// </summary>
+    public class StdioServerProcess
+        : ServerProcess
+    {
+        /// <summary>
+        ///     A <see cref="ProcessStartInfo"/> that describes how to start the server.
+        /// </summary>
+        readonly ProcessStartInfo _serverStartInfo;
+
+        /// <summary>
+        ///     The current server process (if any).
+        /// </summary>
+        Process _serverProcess;
+
+        /// <summary>
+        ///     Create a new <see cref="StdioServerProcess"/>.
+        /// </summary>
+        /// <param name="loggerFactory">
+        ///     The factory for loggers used by the process and its components.
+        /// </param>
+        /// <param name="serverStartInfo">
+        ///     A <see cref="ProcessStartInfo"/> that describes how to start the server.
+        /// </param>
+        public StdioServerProcess(ILoggerFactory loggerFactory, ProcessStartInfo serverStartInfo)
+            : base(loggerFactory)
+        {
+            if (serverStartInfo == null)
+                throw new ArgumentNullException(nameof(serverStartInfo));
+
+            _serverStartInfo = serverStartInfo;
+        }
+
+        /// <summary>
+        ///     Dispose of resources being used by the launcher.
+        /// </summary>
+        /// <param name="disposing">
+        ///     Explicit disposal?
+        /// </param>
+        protected override void Dispose(bool disposing)
+        {
+            if (disposing)
+            {
+                Process serverProcess = Interlocked.Exchange(ref _serverProcess, null);
+                if (serverProcess != null)
+                {
+                    if (!serverProcess.HasExited)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/TylerLeonhardt/PowerShellEditorServices/pullRequest?prid=5596138)